pkg/cloudprovider/aws/suite_test.go

```diff
@@ -370,6 +370,50 @@ var _ = Describe("Allocation", func() {
 					Expect(resources.Pods().Value()).ToNot(BeNumerically("==", 110))
 				}
 			})
+			Context("AL2", func() {
+				It("should calculate memory overhead based on eni limited pods when ENI limited", func() {
+					opts.AWSENILimitedPodDensity = true
+					opts.VMMemoryOverhead = 0 // cutting a factor out of the equation
+					provider.AMIFamily = &v1alpha1.AMIFamilyAL2
+					instanceInfo, err := instanceTypeProvider.getInstanceTypes(ctx, provider)
+					Expect(err).To(BeNil())
+					it := NewInstanceType(injection.WithOptions(ctx, opts), instanceInfo["m5.xlarge"], 0, provider, nil)
+					overhead := it.Overhead()
+					Expect(overhead.Memory().String()).To(Equal("3073Mi"))
+				})
+				It("should calculate memory overhead based on eni limited pods when not ENI limited", func() {
+					opts.AWSENILimitedPodDensity = false
+					opts.VMMemoryOverhead = 0 // cutting a factor out of the equation
+					provider.AMIFamily = &v1alpha1.AMIFamilyAL2
+					instanceInfo, err := instanceTypeProvider.getInstanceTypes(ctx, provider)
+					Expect(err).To(BeNil())
+					it := NewInstanceType(injection.WithOptions(ctx, opts), instanceInfo["m5.xlarge"], 0, provider, nil)
+					overhead := it.Overhead()
+					Expect(overhead.Memory().String()).To(Equal("3073Mi"))
+				})
+			})
+			Context("Bottlerocket", func() {
+				It("should calculate memory overhead based on eni limited pods when ENI limited", func() {
+					opts.AWSENILimitedPodDensity = true
+					opts.VMMemoryOverhead = 0 // cutting a factor out of the equation
+					provider.AMIFamily = &v1alpha1.AMIFamilyBottlerocket
+					instanceInfo, err := instanceTypeProvider.getInstanceTypes(ctx, provider)
+					Expect(err).To(BeNil())
+					it := NewInstanceType(injection.WithOptions(ctx, opts), instanceInfo["m5.xlarge"], 0, provider, nil)
+					overhead := it.Overhead()
+					Expect(overhead.Memory().String()).To(Equal("3073Mi"))
+				})
+				It("should calculate memory overhead based on max pods when not ENI limited", func() {
+					opts.AWSENILimitedPodDensity = false
+					opts.VMMemoryOverhead = 0 // cutting a factor out of the equation
+					provider.AMIFamily = &v1alpha1.AMIFamilyBottlerocket
+					instanceInfo, err := instanceTypeProvider.getInstanceTypes(ctx, provider)
+					Expect(err).To(BeNil())
+					it := NewInstanceType(injection.WithOptions(ctx, opts), instanceInfo["m5.xlarge"], 0, provider, nil)
+					overhead := it.Overhead()
+					Expect(overhead.Memory().String()).To(Equal("1665Mi"))
```

Comment:
First I was confused how we aren't limited by ENIs yet our overhead for BR dropped from `3073Mi` to `1665Mi`. 
I think it makes sense now since it should rather be an increase from `1093Mi` to `1665Mi`.